data_import/serializers.py

```diff
@@ -8,10 +11,33 @@ class DataFileSerializer(serializers.ModelSerializer):
     Serialize a data file.
     """
 
-    download_url = serializers.CharField(source='private_download_url')
     metadata = serializers.JSONField()
 
     class Meta:  # noqa: D101
         model = DataFile
         fields = ('id', 'basename', 'created', 'download_url', 'metadata',
                   'source')
+
+    def to_representation(self, instance):
```

Comment:
Is it possible to...
1. override `_readable_fields` to `super()` then remove `download_url` from the returned set
2. super() `to_representation`, then add `download_url`
?

Thinking about how to do this in a way that minimizes customization on REST framework :)

Reply:
The docs do specifically state:
There are four methods that can be overridden, depending on what functionality you want the serializer class to support:

    .to_representation() - Override this to support serialization, for read op

So, doing it this way seems to be more in line with the way that rest-framework wants things done?

Reply:
But this code seems like it's a copy/paste of some version of `ModelSerializer`... but not the current `master`. And it's relying on `_to_representation` having a particular behavior, so... I don't think this makes less assumptions than my suggestion.

Anything that is copy/paste, and especially when the source it lifted from is changing, troubles me -- my preference to actually defer to whatever REST framework is doing, with a minimal override, not copy/paste it.

Reply:
Obsoleting this; the function is rewritten from scratch now :+1: